File: daehwa_park/4주차/PGS_귤_고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    
    h = Counter(tangerine)
    
    sum = 0
    for i in sorted(h.values(), reverse=True):
        sum += i
        answer += 1
        if sum >= k:
            break
    
    return answer

File: sylee723/16주차/PGS_귤_고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < tangerine.length; i++) {
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0) + 1);
        }
        
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int key : map.keySet()) {
            pq.add(map.get(key));
        }
        
        int total = 0;
        int answer = 0;
        
        while (total < k) {
            int count = pq.poll();
            total += count;
            answer++;
        }
        
        return answer;
    }
}